app/controllers/incomeSources/add/BusinessAccountingMethodController.scala

```diff
@@ -53,26 +54,36 @@ class BusinessAccountingMethodController @Inject()(val authenticate: Authenticat
                                                    val itvcErrorHandlerAgent: AgentItvcErrorHandler)
   extends ClientConfirmedController with FeatureSwitching with I18nSupport {
 
+  def doAllBusinessesCashOrAccrualsMatch(implicit user: MtdItUser[_]): Unit = {
+    if (user.incomeSources.businesses.filterNot(_.isCeased).flatMap(_.cashOrAccruals).distinct.size > 1) {
+    logger.warn("user businesses had different cashOrAccruals types")
+    }
+  }
+
   def handleRequest(isAgent: Boolean)
                    (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext, messages: Messages): Future[Result] = {
 
     val incomeSourcesEnabled: Boolean = isEnabled(IncomeSources)
-    val backUrl: String = if (isAgent) controllers.incomeSources.add.routes.CheckBusinessDetailsController.showAgent().url else
-      controllers.incomeSources.add.routes.CheckBusinessDetailsController.show().url
+    val backUrl: String = if (isAgent) controllers.routes.AddBusinessAddressController.showAgent().url else
+      controllers.routes.AddBusinessAddressController.show().url
     val postAction: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessAccountingMethodController.submitAgent() else
       controllers.incomeSources.add.routes.BusinessAccountingMethodController.submit()
 
     val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
 
     if (incomeSourcesEnabled) {
-      val userBusinesses: List[BusinessDetailsModel] = user.incomeSources.businesses
-      if (shouldAutomaticallyRedirect(userBusinesses)) {
+
+      doAllBusinessesCashOrAccrualsMatch(user)
+
+      val userActiveBusinesses: List[BusinessDetailsModel] = user.incomeSources.businesses.filterNot(_.isCeased)
+      if (!userActiveBusinesses.forall(_.isCeased)) {
+        val accountingMethod: String = userActiveBusinesses.head.cashOrAccruals.getOrElse(throw MissingFieldException("cashOrAccruals field missing"))
```

Comment:
you are not handling error in the right way; in case this would happen, you should not raise error